src/firejail/main.c

```diff
@@ -1498,8 +1498,18 @@ int main(int argc, char **argv, char **envp) {
 			arg_rlimit_sigpending = 1;
 		}
 		else if (strncmp(argv[i], "--rlimit-as=", 12) == 0) {
-			check_unsigned(argv[i] + 12, "Error: invalid rlimit");
-			sscanf(argv[i] + 12, "%llu", &cfg.rlimit_as);
+			char last_char = *(argv[i]+strlen(argv[i])-1);
+			if( !isdigit(last_char) ){
+				if( is_valid_unit(last_char) ){
+					sscanf(argv[i]+12,"%llu", &cfg.rlimit_as);
+					cfg.rlimit_as = unit_to_byte(last_char,cfg.rlimit_as);
+				} else {
+					check_unsigned(argv[i]+12,"Error: invalid rlimit size specifier. only k,m,g supported ");
```

Comment:
hm, in this error case we probably want to `exit()`, or?
(also in `profile.c`)

Reply:
I mean, exit directly, no need to check if the value is a valid unsigned number.

Reply:
i thought same thing but i don't know codebase very well. so i just used same exit strategy in case of some cleanup occurs in these functions. 

( yes, i am bit lazy )

Reply:
ok, let's keep it.